src/assumptions.h

```diff
@@ -19,7 +21,11 @@ struct secp256k1_assumption_checker {
        allowed. */
     int dummy_array[(
         /* Bytes are 8 bits. */
-        CHAR_BIT == 8 &&
+        (CHAR_BIT == 8) &&
+
+        /* No integer promotion for uint32_t. This ensures that we can multiply uintXX_t values where XX >= 32
+           without signed overflow, which would be undefined behaviour. */
+        (UINT_MAX <= UINT32_MAX) &&
```

Comment:
I think `INT_MAX < UINT32_MAX` would more directly capture your intention here; however given the other rules of C, I do admit that what you have written ends up being equivalent.

Reply:
Yeah, I think our intuition is just different. My thought was that if `uint32_t` is at least as large as `unsigned int`, then no promotion can happen at all, neither to `signed int` nor to `unsigned int`.

Reply:
I think of the rule as if every value of a type can fit within a `signed int`, then it is subject to integer promotion.

Reply:
Both seem reasonable and harmless to me.

Reply:
Wanna ACK then?